asyncstatsd/client/basic.py

```diff
@@ -0,0 +1,81 @@
+import contextlib
+import random
+from abc import ABC, abstractmethod
+from datetime import timedelta
+from time import perf_counter as time_now
+
+from ..metrics.basic import (
+    CounterMetric,
+    GaugeMetric,
+    SetMetric,
+    StatsdMetric,
+    TimingMetric,
+)
+from .udp import UDPClient
+
+
+class StatsdClientBase(UDPClient):
```

Comment:
I'd consider DI instead inheritance
Though it may not be very common, for some customizing or using other transport layer might be an advantage